test/e2e/azure_suite_test.go

```diff
@@ -0,0 +1,191 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_test
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/config"
+	"github.com/onsi/ginkgo/reporters"
+	. "github.com/onsi/gomega"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	bootstrapv1 "sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/api/v1alpha2"
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha2"
+	"sigs.k8s.io/cluster-api-provider-azure/test/e2e/auth"
+	"sigs.k8s.io/cluster-api-provider-azure/test/e2e/framework"
+	"sigs.k8s.io/cluster-api-provider-azure/test/e2e/framework/management/kind"
+	"sigs.k8s.io/cluster-api-provider-azure/test/e2e/generators"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1alpha2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestE2E(t *testing.T) {
+	artifactPath, _ := os.LookupEnv("ARTIFACTS")
+	junitXML := fmt.Sprintf("junit.e2e_suite.%d.xml", config.GinkgoConfig.ParallelNode)
+	junitPath := path.Join(artifactPath, junitXML)
+	junitReporter := reporters.NewJUnitReporter(junitPath)
+
+	RegisterFailHandler(Fail)
+	RunSpecsWithDefaultAndCustomReporters(t, "CAPZ e2e suite", []Reporter{junitReporter})
+}
+
+var (
+	ctx   = context.Background()
+	creds auth.Creds
+	mgmt  *kind.Cluster
+
+	// TODO Parameterize some of these variables
+	location       = "westus2"
+	vmSize         = "Standard_B2ms"
+	namespace      = "default"
+	k8sVersion     = "v1.16.2"
+	imageOffer     = "capi"
+	imagePublisher = "cncf-upstream"
+	imageSKU       = "k8s-1dot16-ubuntu-1804"
+	imageVersion   = "latest"
+)
+
+var _ = BeforeSuite(func() {
+	var err error
+
+	By("Loading Azure credentials")
+	if credsFile, found := os.LookupEnv("AZURE_CREDENTIALS"); found {
+		creds, err = auth.LoadFromFile(credsFile)
+	} else {
+		creds, err = auth.LoadFromEnvironment()
+	}
+	Expect(err).NotTo(HaveOccurred())
+	Expect(creds).NotTo(BeNil())
+	Expect(creds.TenantID).NotTo(BeEmpty())
+	Expect(creds.SubscriptionID).NotTo(BeEmpty())
+	Expect(creds.ClientID).NotTo(BeEmpty())
+	Expect(creds.ClientSecret).NotTo(BeEmpty())
+
+	By("Creating management cluster")
+	scheme := runtime.NewScheme()
+	Expect(appsv1.AddToScheme(scheme)).To(Succeed())
+	Expect(corev1.AddToScheme(scheme)).To(Succeed())
+	Expect(capiv1.AddToScheme(scheme)).To(Succeed())
+	Expect(bootstrapv1.AddToScheme(scheme)).To(Succeed())
+	Expect(infrav1.AddToScheme(scheme)).To(Succeed())
+
+	managerImage, found := os.LookupEnv("MANAGER_IMAGE")
+	Expect(found).To(BeTrue(), fmt.Sprint("MANAGER_IMAGE not set"))
+
+	mgmt, err = kind.NewCluster(ctx, "mgmt", scheme, managerImage)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(mgmt).NotTo(BeNil())
+
+	// TODO Figure out how to keep these versions in sync across the code base
```

Comment:
can you open an issue to track this please?

Reply:
added [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/339#issuecomment-564200720)

Reply:
added to the list above